libdnf/dnf-keyring.cpp

```diff
@@ -136,6 +138,20 @@ dnf_keyring_add_public_key(rpmKeyring keyring,
         goto out;
     }
 
+    subkeys = rpmGetSubkeys(pubkey, &nsubkeys);
+    for (gint i = 0; i < nsubkeys; i++) {
+        rpmPubkey subkey = subkeys[i];
+        if (rpmKeyringAddKey(keyring, subkey) < 0) {
```

Comment:
In DNF only subkeys that pass the test `subkey.can_sign` are added. Please could you explain what exactly the code should do.

Reply:
This patch essentially matches what librpm does (see `loadKeyringFromFiles()`). In a quick scan, it's not clear whether librpm makes the same distinction. If not, I think this is something we'd need to fix there anyway, right? Is there anyone from the rpm team we can check with?

Reply:
@jlebon I looked into librpm. And yes, librpm does the same as this patch. I've been thinking about it. I will merge the PR. There is a plan to create GPG API in libdnf. So I hope to clarification during this work.

Reply:
Great, WFM!

Reply:
@jlebon Wait? You want to make other changes to PR? Please set Labels to `ready for review` when the PR is ready.

Reply:
Ahh, I meant Works For Me :)

I don't have access to set labels, but the PR is ready.

Reply:
I read `Wait For Me` instead of `Works for me` :-)